### PR TITLE
Add Leanplum integration

### PR DESCRIPTION
**Pull request by rmi22186**
Initial commit for Leanplum Integration

### CODE REVIEW

File: test/tests.js [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    it('it should set user identity if IdentityType = CustomerId', function(done) {
+        mParticle.forwarder.setUserIdentity('123abc', IdentityType.CustomerId);
+
+        window.Leanplum.userId.should.equal('123abc');
+
+        done();
+    });
+
+    it('it should set userId of type customerId if user identities passed to it include types customerId and email', function(done) {
+        window.Leanplum.userId.should.equal('customerId');
```
Comment (tbreffni) [inline test/tests.js]: Minor - test relies on previous test having executed. I think it's usually better if unit tests are independent of one another.

File: LeanplumAnalyticsEventForwarder.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        }
+
+        function completeLeanPlumInitialization(forwarderSettings, userIdentities) {
+            setLeanPlumEnvironment(forwarderSettings);
+            initializeUserId(userIdentities);
+            isInitialized = true;
+        }
+
+        function setLeanPlumEnvironment(forwarderSettings) {
+            if (window.mParticle.isDebug) {
```
Comment (samdozor) [inline LeanplumAnalyticsEventForwarder.js]: I think this should be isSandbox

File: test/tests.js (hunk 1 above)
Comment (rmi22186) [inline test/tests.js]: ah - this one depends on the "beforeEach" function calls when userIdentities is being passed to `init`. to confirm, i isolated the test with `it.only()` and it still worked

**Comment by rmi22186** [comment]
Fixed above comments. Good for another 👀

**Pull request merged**